Add clarification about schedule_to_start (#858)
temporalio/documentation: docs/content/what-is-a-schedule-to-start-timeout.md

import CenteredImage from "../components/CenteredImage.js"
import RelatedReadList from '../components/RelatedReadList.js'

A Schedule-To-Start Timeout is the maximum amount of time that is allowed from when an [Activity Task](/docs/content/what-is-an-activity-task) is scheduled (that is, placed in a Task Queue) to when a [Worker](/docs/content/what-is-a-worker) starts (that is, picks up from the Task Queue) that Activity Task.
In other words, it's a limit for how long an Activity Task can be enqueued.

> The moment that the Task is picked by the Worker from the task queue is considered to be the start of the Activity Task for the Schedule To Start timeout and associated metrics to avoid clock difference between the Server and the Worker.

Reviewer: ```suggestion
The moment that the Task is picked by the Worker from the Task Queue is considered to be the start of the Activity Task for the purposes of the Schedule-To-Start timeout and associated metrics to avoid clock difference between the Server and the Worker.
```

import CenteredImage from "../components/CenteredImage.js"
import RelatedReadList from '../components/RelatedReadList.js'

> A Schedule To Start Timeout is the maximum amount of time that is allowed, from when an [Activity Task](/docs/content/what-is-an-activity-task) is scheduled (placed in a Task Queue) to when a [Worker](/docs/content/what-is-a-worker) starts (picks it up from the Task Queue) that Activity Task.
> In other words, it's a limit for how long an Activity Task can be enqueued.
> The moment that the Task is picked by the Worker from the task queue is considered to be the start of the Activity Task for the Schedule To Start timeout and associated metrics to avoid clock difference between the Server and the Worker.

Reviewer: ```suggestion
A Schedule-To-Start Timeout is the maximum amount of time that is allowed, from when an [Activity Task](/docs/content/what-is-an-activity-task) is scheduled (placed in a Task Queue) to when a [Worker](/docs/content/what-is-a-worker) starts executing that Activity Task (picks the Activity Task from the Task Queue).
In other words, a Schedule-To-Start Timeout is a limit on how long an Activity Task can remain in the Task Queue.
The moment the Activity Task is picked by the Worker is considered to be the start of the Activity Task Execution for the purposes of calculating the Schedule-To-Start Timeout and associated metrics.
It's designed this way in case there are clock differences between the Temporal Cluster and a Worker.
```
I personally liked it before and am advocating for you to change it back.
I do not think "It's" is too colloquial here as "It's" clearly refers to the dynamic described in the previous sentence - @djmagee what is your take on this paragraph?

Reviewer 2: The avatar of one of the sharpest project editors I know says that Shawn is correct here. (She always insisted that I crisply define antecedents.)

I recommend changing your last sentence to something like this:

This definition of "start" avoids issues that a clock difference between the Temporal Cluster and a Worker might create.

<CenteredImage
imagePath="/diagrams/schedule-to-start-timeout.svg"
imageSize="100"
title="Schedule-To-Start Timeout period"
/>

> A [Retry Policy](/docs/content/what-is-a-retry-policy) attached to an Activity Execution retries an Activity Task.
> Thus, the Schedule-To-Start Timeout and latency metrics are applied to each Activity Task separately within an Activity Execution.
> 
> :::warning "Schedule" in Schedule-To-Start and Schedule-To-Close may reference different moments
> 
> "Schedule" in Schedule-To-Start references the scheduling of a single Activity Task (each Activity retry has a separate scheduling moment).
> "Schedule" in Schedule-To-Close references the scheduling of the whole Activity Execution.
> 
> :::

Reviewer: ```suggestion
"Schedule" in Schedule-To-Start and Schedule-To-Close both refer to the scheduling of an Activity Task.
Schedule-To-Close refers to the scheduling of the **first** Activity Task in the set of Activity Tasks that make up the Activity Execution.
"Schedule" in Schedule-To-Start refers to the scheduling of **any** Activity Task in the set of Activity Tasks that make up the Activity Execution.
In other words, Schedule-To-Close Timeout is enforced once per Activity Execution, whereas Schedule-To-Start Timeout is enforced for each Activity Task.
This is because a [Retry Policy](/docs/content/what-is-a-retry-policy) attached to an Activity Execution retries an Activity Task.
Thus, the Schedule-To-Start Timeout and latency metrics are applied to each Activity Task separately within an Activity Execution.
```
Let's stay away from admonitions if we can.

@Spikhalskiy perhaps "enforced" is the wrong word here? But just trying to synthesize the point a bit and build on existing information that aims to touch on this point.
The whole thing is slightly verbose now, but hits it from all angles.
We can add some of @dtornow diagrams in a future iteration to help hammer things in.

One thing though is, If we call it out here, then we should make the similar point in the Schedule-To-Close page.

Reviewer 2: Agreed. I recommend changing the bold \*\*first\*\* and \*\*any\*\* to italic (\*first\* and \*any\*) to match the style guide's guidance on emphasis (https://developers.google.com/style/text-formatting).

Author: "Enforced" looks pretty correct for me here.

Author: @djmagee addressed the "italic" change

<CenteredImage
imagePath="/diagrams/schedule-to-start-timeout-with-retry.svg"
imageSize="100"
title="Start-To-Close Timeout period with retries"
/>

> There are two primary use cases of this timeout:

Reviewer 2: ```suggestion
This timeout has two primary use cases:
```

import CenteredImage from "../components/CenteredImage.js"
import RelatedReadList from '../components/RelatedReadList.js'

A Schedule To Start Timeout is the maximum amount of time that is allowed, from when an [Activity Task](/docs/content/what-is-an-activity-task) is scheduled (placed in a Task Queue) to when a [Worker](/docs/content/what-is-a-worker) starts (picks it up from the Task Queue) that Activity Task.

Thus, the moment that the Task is picked by the Worker is considered to be the start of the Activity Task for the Schedule To Start timeout and associated metrics.

> It's designed this way to avoid clock difference between the Server and the Worker.

Reviewer 3: “It’s” contraction is too colloquial for this docs and often can find a way to be more specific about what “It” is

<CenteredImage
imagePath="/diagrams/schedule-to-start-timeout.svg"
imageSize="100"
title="Schedule-To-Start Timeout period"
/>

A [Retry Policy](/docs/content/what-is-a-retry-policy) attached to an Activity Execution retries an Activity Task.
Thus, the Schedule-To-Start Timeout and latency metrics are applied to each Activity Task separately within an Activity Execution.

> :::warning "Schedule" in Schedule-To-Start and Schedule-To-Close reference potentially different moments

Reviewer 3: nice nuance, would classify these as part of an “advanced” section to signify that most users wont need to know this

Author: If they are reading this doc, they do though. There were probably two main things that the reader needs to know in Schedule-To-Start timeout.
What specifically is Schedule here and don't mix it with another Schedule and what is considered Start.

import CenteredImage from "../components/CenteredImage.js"
import RelatedReadList from '../components/RelatedReadList.js'

A Schedule-To-Start Timeout is the maximum amount of time that is allowed from when an [Activity Task](/docs/content/what-is-an-activity-task) is scheduled (that is, placed in a Task Queue) to when a [Worker](/docs/content/what-is-a-worker) starts (that is, picks up from the Task Queue) that Activity Task.
In other words, it's a limit for how long an Activity Task can be enqueued.

> The moment that the Task is picked by the Worker from the task queue is considered to be the Start of the Activity Task for the Schedule To Start timeout and associated metrics.

Reviewer: ```suggestion
The moment that the Task is picked by the Worker from the Task Queue is considered to be the Start of the Activity Task for the purposes of the Schedule-To-Start Timeout and associated metrics.
```

This definition of "Start" avoids issues that a clock difference between the Temporal Cluster and a Worker might create.

<CenteredImage
imagePath="/diagrams/schedule-to-start-timeout.svg"
imageSize="100"
title="Schedule-To-Start Timeout period"
/>

> :::warning "Schedule" in Schedule-To-Start and Schedule-To-Close may reference different moments
> 
> Schedule-To-Start Timeout is enforced for each Activity Task, whereas Schedule-To-Close Timeout is enforced once per Activity Execution.
> Thus, "Schedule" in Schedule-To-Start refers to the scheduling moment of *every* Activity Task in the sequence of Activity Tasks that make up the Activity Execution, while
> "Schedule" in Schedule-To-Close refers to the *first* Activity Task in that sequence.
> 
> :::

Reviewer: ```suggestion
"Schedule" in Schedule-To-Start and Schedule-To-Close have different frequency guarantees.

The Schedule-To-Start Timeout is enforced for each Activity Task, whereas the Schedule-To-Close Timeout is enforced once per Activity Execution.
Thus, "Schedule" in Schedule-To-Start refers to the scheduling moment of *every* Activity Task in the sequence of Activity Tasks that make up the Activity Execution, while
"Schedule" in Schedule-To-Close refers to the *first* Activity Task in that sequence.
```

Like I said before, let's stay away from the admonitions - this is perfectly suitable core information.
